### DISCUSSION TITLE
How to run specific migration or migrations in specific order?

### DISCUSSION THREAD

**Discussion by kyleerhabor** [diesel-rs/diesel#2751] [Q&A] [answered]
I'm using the Diesel CLI to manage tables. I have a `users` table that references the `images` table with foreign keys. I don't see how I'd be able to run a specific migration or run the migration in a specific order. The `users` table is being run first when I run `diesel migration run`, and since the `images` table doesn't exist yet, it fails, saying `relation "images" does not exist`.

**Comment by weiznich** [comment, marked as answer]
Diesel's migrations are inherently ordered, by the order you've created the migrations via `diesel migration generate` (or technically speaking by the part of the migration directory before the first `_`.)
So if you have a migration that depends on a other migration just put it into a migration with a larger version that the one it depends on. See the [documentation of the diesel_migrations](https://docs.diesel.rs/1.4.x/diesel_migrations/index.html) crate for details.

**Reply by kyleerhabor** [reply]
Thanks for the response. I decided to prefix migrations with a number (e.g. `v1`) to make the ordering easier, which solved my problem.
